### PR TITLE
feat:Analytics and rebranding

### PR DESCRIPTION
**Pull request by Aman-Aalam**
# Feature #

Added `Trolley-Source` header in requests, added `routeMinimum` attribute, rebranded `Payment Rails` instances to `Trolley` wherever possible.

### Checklist

- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

If you have questions, create a GitHub Issue in this repository.

### CODE REVIEW

**Comment by Aman-Aalam** [comment]
Added tests to make sure `routeMinimum` functionality works

File: docs/classes/configuration.md [hunk 1]
```diff
@@ -14,11 +14,11 @@
...
 ## Properties
 
-<a id="enviroment"></a>
+<a id="environment"></a>
 
-### enviroment
+### environment
 
-**●  enviroment**:  *`string`*
+**●  environment**:  *`string`*
```
Comment (pragati-l) [inline docs/classes/configuration.md:21]: Wow good eyes ;)

File: docs/classes/configuration.md [hunk 2]
```diff
@@ -52,7 +52,7 @@
 
 *Defined in [Configuration.java:90](https://github.com/PaymentRails/java-sdk/tree/master/src/main/java/ca/paymentrails/paymentrails/Configuration.java#L90)*
 
-Set the Payment Rails API environment that your using
+Set the Trolley API environment that your using
```
Comment (pragati-l) [inline docs/classes/configuration.md]: your using?

**Review by pragati-l — APPROVED** [review]
LGTM

**Pull request merged**